### PR TITLE
JBIDE-25430 - create choose openshift connection composite, fragment,…

### PR DESCRIPTION
**Pull request by robstryker**
… etc

Signed-off-by: Rob Stryker <rob@oxbeef.net>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionFragment.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	
+	private WizardDialog getContainer(IWizardPage wizardPage) {
+		return (WizardDialog) wizardPage.getWizard().getContainer();
+	}
+
+	public void addConnectionChangeListener(PropertyChangeListener connectionChangeListener) {
+		this.externalConnectionListener = connectionChangeListener;
+	}
+	
+	private IPageChangingListener onPageChanging(IWizardHandle wizardHandle) {
```
[the remark this answers is not in the document]
Comment (robstryker) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ChooseOpenshiftConnectionFragment.java]: I can't reduce this any more without it being ridiculous.

**Comment by adietish** [comment]
@robstryker thanks for the idea to change this. I like the idea to separate editing and choosing a connection even though I believe that there are glitches:

1. with this patch we now have the approach where you get a dialog on top of the existing wizard, which is a pattern one should avoid.
2. If you hit "New" you get into the existing wizard and can choose an existing connection, edit a connection etc. which is not logical (and you get the dialog on top of dialog pattern). Equally if you hit "Edit" you can create a new connection.

IMHO this basically shows that there's no way around changing the existing connection wizard. In the above usecases one is only allowed to edit an existing connection, not edit it (you hit "Edit") or create a new connection, not edit it (you hit "New"). 

Other glitches:
1. Whatever you do (either create a new connection or edit an existing connection, the connection that you have when you hit "Finish" isn't selected when you get back to the parent (server-) wizard.
2. There's no connection preselected if when you launch the server wizard File > New > Server > OpenShift 3 Server Adapter

Here are a few UI glitches that I identified:
image

**Comment by robstryker** [comment]
I've fixed a lot of the UI issues you had. I don't think we should demand the connection wizard itself be fixed as part of this jira, but rather as a next item on the agenda.

**Comment by adietish** [comment]
+1 for the new version of the PR as soon as:
- spacing btw. server and "Authentication" goup is increased
- initial enablement of "Next" is fixed.

**Pull request merged**